### PR TITLE
support for @angular/forms

### LINKED ISSUE #627: feat: Support the RC.2 Forms
We should work on a version that supports the new forms: https://docs.google.com/document/u/1/d/1RIezQqE4aEhBRmArIAS1mRIZtWFf6JxN_7B4meyWK0Y/pub

Currently, when I disable the deprecated forms, some of the ng2-bootstrap components are not able to find NgModel and other stuff (for example in typeaheads).

### PR DESCRIPTION
**Pull request by jFensch**
This PR should resolve #627.  Uses the new providers/directives from @angular/forms

### CODE REVIEW

**Comment by Dinistro** [comment]
thank you very much. I will take a look at this later today.

**Comment by Dinistro** [comment]
@jFensch You forgot to add the @angular/forms to the .config/bundles-system.js.Can you please update this? 

@valorkin This contains breaking changes. We need to make sure, that we release these changes in a 1.2.\* version.

File: package.json [hunk 1]
```diff
@@ -50,17 +50,19 @@
...
+    "@angular/forms": "^0.1.0"
   },
   "devDependencies": {
-    "@angular/common": "^2.0.0-rc.1",
-    "@angular/compiler": "^2.0.0-rc.1",
-    "@angular/core": "^2.0.0-rc.1",
-    "@angular/platform-browser": "^2.0.0-rc.1",
-    "@angular/platform-browser-dynamic": "^2.0.0-rc.1",
-    "@angular/router-deprecated": "2.0.0-rc.1",
+    "@angular/common": "^2.0.0-rc.2",
```
Comment (Dinistro) [inline package.json:61]: I get some warnings here, but when I changed the versions to rc.3 (except router-deprecated) it works fine.

**Comment by valorkin** [comment]
@Dinistro and rebase is must, or cherry pick at least

**Comment by Dinistro** [comment]
@jFensch Can you please rebase your branch to resolve the conflicts?

**Comment by Dinistro** [comment]
@jFensch Thanks for the rebase but unfortunately some tests are not working, can you please fix them?

**Comment by Dinistro** [comment]
@valorkin Now we need to do this by our own. Unluckily I have finals on monday and so I don't have time to do it right now.

**Comment by jFensch** [comment]
Recloning and fixing the tests...give me a minute.

Sorry

**Comment by kenkopelson** [comment]
Hi guys :)  Any idea when the broken 1.0.17 will be fixed?  I'm getting the "No provider for NgModel" also, and can't move forward with my development.  Would really appreciate this being fixed.

**Comment by trollr** [comment]
+1

**Comment by mpx200** [comment]
i'm having same issue with rc4

**Comment by jprinaldi** [comment]
Yeah, same here. Can't wait for the update.

Thanks for your hard work as always guys!

**Comment by kenkopelson** [comment]
Yes, thank you for the hard work and commitment.  All of us who are using the ng-bootstrap library really do appreciate it.  We would not be able to move forward with Angular 2 without it!  Please let us know about the status of this breakage so we can make plans accordingly.  Much appreciated :)

**Comment by valorkin** [comment]
In order to use new forms you need to enable them:
https://github.com/valor-software/ng2-bootstrap/blob/development/demo/index.ts#L57-L58
and use NgModel from angular/forms

**Comment by trollr** [comment]
Don't work for me. Added 

`disableDeprecatedForms(),
 provideForms(),`

and used NgModel from @angular/forms. Still `ORIGINAL EXCEPTION: No provider for NgModel!` with ng2-bootstrap 1.0.20 for typeahead

**Comment by Dinistro** [comment]
@trollr @valorkin i will take a look at this.

**Comment by jprinaldi** [comment]
Just a heads up, it does work fine for me with 1.0.22.

**Comment by Dinistro** [comment]
@trollr @juampi I forgot to add the support for typeahead inside a form...

**Comment by jprinaldi** [comment]
Oh, I see. I didn't test typeahead.

**Comment by Dinistro** [comment]
@trollr This PR should fix it: #723

**Pull request closed without merging**